### DISCUSSION TITLE
Argument notation

### DISCUSSION THREAD

**Discussion by webyrd** [ekzhang/percival#5] [Q&A] [answered]
Hi Eric!

```
// Given an edge x -> y, there is a path x -> y.
path(x, y) :- edge(x, y).

// Given an edge x -> z and a path z -> y, there is a path x -> y.
path(x, y) :- edge(x, y: z), path(x: z, y).
```

I was confused by `edge(x, y: z)`, which seems to match positional argument notation with named arguments.  What are the rules for combining named arguments with positional arguments in a call to a relation?

Is this standard Datalog notation, or an extension you added?

Thank you!

Cheers,

--Will

**Comment by ekzhang** [comment, marked as answer]
Hi Will! There's no positional arguments in Percival; all arguments are named. This is to allow us to do queries on named records, as is common in datasets for machine learning, data science, etc.

In this example, `path(x, y)` is a shorthand syntax for `path(x: x, y: y)`.

**Comment by webyrd** [comment]
Thanks, Eric!  :)

I think it would be worth explaining this is the tutorial.

Cheers,

--Will

**Reply by ekzhang** [reply]
Yeah for sure!! Thanks for all the useful feedback :) still have a lot of work to do on the tutorial, haha.
